### DISCUSSION TITLE
How about libraries?

### DISCUSSION THREAD

**Discussion by federicopeyrani1969** [orbat-mapper/orbat-mapper#211] [Ideas]
Hi Magne,
I was thinking if you consider users making and sharing "Unit libraries". Can you emit specifics for that?
For example, I am working on an Access/Excel table listing more than 20000 German WW2 units of all arms from [Lexikon der Wehrmacht](https://www.lexikon-der-wehrmacht.de/). Don't you think it could be useful for others?
I made it because often German units changed denomination maintaining the numbering, and this was reported only later on maps, 
I was becoming crazy trying to understand the correct denomination of the map icons.
Best Regards
Federico

**Comment by kjellmf** [comment]
A form of unit library is a good idea, but I don't know what it should look like. What kind of data (columns) do you store in your table?

**Reply by federicopeyrani1969** [reply]
My files had two purposes: finding an unit by its number/name and (as a future intention) understanding unit genealogy.
I'll send you an excel copy of my project via private mail in a couple of minutes. Units are 20757 at the moment, but I realized that I duplicated some because there are more ways to get to their list.

**Reply by federicopeyrani1969** [reply]
Please note I'm an history fan, not a logic fan, so I tried to make it useful for the task but I don't know if I'm wrong somewhere..

Fields were as from this image: 
![2](https://github.com/orbat-mapper/orbat-mapper/assets/58425768/91190694-9482-43e4-afe0-1f7b701f17c4)
"Number" field was for quick searching and listing, 
So e.g. for **9. Panzer-Division** was "**00009**" (as for 9. Infanterie-Division) and for "**Herman Göring**" units was "**H-Namen**".
Hope it is clear enough.
Regards
Federico

**Comment by federicopeyrani1969** [comment]
I still have one/two question/hint about libraries.

After the "Unit Libraries", now it's the turn of another kind of Libraries:  
**"Vehicle Libraries"**, maybe divided by conflict/period.
I mean: if I have to display a tactical level scenery, I could pick a specific tank (e.g. Sherman, Panzer III, Tiger, T-72, M1 Abrams, Leopard 2), and the system/library then gives the right Stanag symbol for it. Wouldn't it be Interesting?

A third kind of Library comes now to my mind: it could be **"TOE Librairies"**, so that I can refer to a theoretical TOE composition putting my unit on the ground. The German KStN tables are well known and easily replicable. I can supply a book listing them or the link to sites like Nafziger, full of TOE tables, at least for WWII.

Thank you for your patience!
Federico

**Reply by kjellmf** [reply]
Thank you for your ideas. This is certainly possible, but maintaining a large equipment and TOE library is not something I plan to do. I probably could add support for loading data from such a library though. For modern day equipment there is a library available at https://odin.tradoc.army.mil/WEG There is also https://www.matrixgames.com/game/command-modern-operations

**Reply by federicopeyrani1969** [reply]
What about putting somewhere in the site some templates for a library? (of Units-Vehicles-TOEs).
I think a blank file of whatever format (xls/docx/txt/csv) with specific fields for the importation.
This would allow people like to make/upload/share their work, for importing/uploading libraries when needed.
Regards
Federico
